tests/test_template_types.py

```diff
@@ -618,26 +626,89 @@ def test_templates_handle_html_and_redacting(
             "<span class='placeholder'>((postcode))</span>"
         ),
         mock.call(Markup('subject')),
-        mock.call(Markup('content')),
+        mock.call(Markup('content<cr><cr>')),
         mock.call(Markup('subject')),
         mock.call(Markup('subject')),
     ]),
 ])
-@mock.patch('notifications_utils.template.remove_whitespace_before_commas', side_effect=lambda x: x)
-def test_templates_handle_html_and_redacting(
-    mock_remove_commas,
+@mock.patch('notifications_utils.template.remove_whitespace_before_punctuation', side_effect=lambda x: x)
+def test_templates_remove_whitespace_before_punctuation(
+    mock_remove_whitespace,
+    template_class,
+    extra_args,
+    expected_remove_whitespace_calls,
+):
+    template = template_class({'content': 'content', 'subject': 'subject'}, **extra_args)
+
+    assert str(template)
+
+    if hasattr(template, 'subject'):
+        assert template.subject
+
+    assert mock_remove_whitespace.call_args_list == expected_remove_whitespace_calls
+
+
+@pytest.mark.parametrize('template_class, extra_args, expected_calls', [
+    (PlainTextEmailTemplate, {}, [
+        mock.call('content'),
+        mock.call(Markup('subject')),
+        mock.call(Markup('subject')),
+    ]),
+    (HTMLEmailTemplate, {}, [
+        mock.call(
+            '<p style="Margin: 0 0 20px 0; font-size: 19px; line-height: 25px; color: #0B0C0C;">'
+            'content'
+            '</p>'
+        ),
+        mock.call(Markup('subject')),
+        mock.call(Markup('subject')),
+    ]),
+    (EmailPreviewTemplate, {}, [
+        mock.call(
+            '<p style="Margin: 0 0 20px 0; font-size: 19px; line-height: 25px; color: #0B0C0C;">'
+            'content'
+            '</p>'
+        ),
+        mock.call(Markup('subject')),
+        mock.call(Markup('subject')),
+        mock.call(Markup('subject')),
+    ]),
+    (SMSMessageTemplate, {}, [
+    ]),
+    (SMSPreviewTemplate, {}, [
+    ]),
+    (LetterPreviewTemplate, {'contact_block': 'www.gov.uk'}, [
+        mock.call(Markup('subject')),
+        mock.call(Markup('<p>content</p>')),
+        mock.call(Markup('subject')),
+        mock.call(Markup('subject')),
+    ]),
+    (LetterDVLATemplate, {'notification_reference': "1", 'contact_block': 'www.gov.uk  '}, [
+        mock.call('subject'),
+        mock.call('content<cr><cr>'),
+        mock.call('subject'),
+        mock.call('subject'),
```

Comment:
These tests all feel very tautological - they test what they test because the code that they run produces the results that it does.

what would happen if tomorrow `str(LetterDVLATemplate)` only called make_quotes_smart with `'subject'` twice instead of three times?